elyra/pipeline/pipeline_definition.py

```diff
@@ -0,0 +1,281 @@
+        for pipeline in self._pipeline_definition['pipelines']:
+            pipelines.append(Pipeline(pipeline))
+
+        return pipelines
+
+    @property
+    def primary_pipeline(self) -> Pipeline:
+        if not self._primary_pipeline:
+            if "pipelines" not in self._pipeline_definition:
+                raise ValueError("Pipeline is missing 'pipelines' field.")
+            elif len(self._pipeline_definition['pipelines']) == 0:
+                raise ValueError("Pipeline has zero length 'pipelines' field.")
+
+            # Find primary pipeline
+            self._primary_pipeline = self._get_pipeline_definition(self._pipeline_definition['primary_pipeline'])
+
+            assert self._primary_pipeline is not None, "No primary pipeline was found"
+
+        return Pipeline(self._pipeline_definition, self._primary_pipeline)
+
+    def validate(self) -> list:
+        validation_issues = list()
+        # Validate pipeline schema version
+        if 'version' not in self._pipeline_definition:
+            validation_issues.append("Pipeline schema version field is missing.")
+        elif not isinstance(self._pipeline_definition['version'], str):
+            validation_issues.append("Pipeline schema version field should be a string.")
+
+        # Validate pipelines
+        if 'pipelines' not in self._pipeline_definition:
+            validation_issues.append("Pipeline is missing 'pipelines' field.")
+        elif not isinstance(self._pipeline_definition["pipelines"], list):
+            validation_issues.append("Field 'pipelines' should be a list.")
+        elif len(self._pipeline_definition['pipelines']) == 0:
+            validation_issues.append("Pipeline has zero length 'pipelines' field.")
+
+        # Validate primary pipeline
+        if 'primary_pipeline' not in self._pipeline_definition:
+            validation_issues.append("Could not determine the primary pipeline.")
+        elif not isinstance(self._pipeline_definition["primary_pipeline"], str):
+            validation_issues.append("Field 'primary_pipeline' should be a string.")
+
+        primary_pipeline = self._get_pipeline_definition(self._pipeline_definition.get('primary_pipeline'))
+        if not primary_pipeline:
+            validation_issues.append("No primary pipeline was found")
+        else:
+            # Validate primary pipeline structure
+            if 'nodes' not in primary_pipeline or len(primary_pipeline['nodes']) == 0:
+                validation_issues.append("At least one node must exist in the primary pipeline.")
+            if 'app_data' not in primary_pipeline:
+                validation_issues.append("Primary pipeline is missing the 'app_data' field.")
+            elif 'version' not in primary_pipeline['app_data']:
+                validation_issues.append("Primary pipeline is missing the 'version' field.")
+
+        return validation_issues
+
+    def is_valid(self) -> bool:
+        return len(self.validate()) == 0
```

Comment:
Again, not sure we want to validate the entire definition just to see if it's valid or not.